src/appleseed/foundation/platform/system.cpp

```diff
@@ -421,6 +429,388 @@ uint64 System::get_process_virtual_memory_size()
     return static_cast<uint64>(rss) * sysconf(_SC_PAGESIZE);
 }
 
+// ------------------------------------------------------------------------------------------------
+// FreeBSD.
+// ------------------------------------------------------------------------------------------------
```

Comment:
Quick question: do I have to reindent all code when I move it inside anonymous namespace?
About class names: so correct name is `TrivialX86Cache`, right?